vite.config.ts

```diff
@@ -1,7 +1,34 @@
-import { defineConfig } from 'vite';
+import fs from 'node:fs';
+import path from 'node:path';
+
 import react from '@vitejs/plugin-react';
+import { splitVendorChunkPlugin } from 'vite';
+import { defineConfig } from 'vitest/config';
+
+const plugins = [react(), splitVendorChunkPlugin()];
+
+const pages = fs.readdirSync(path.join(__dirname, 'pages'));
+
+const rollupOptions = {
+  input: {
+    index: path.join(__dirname, 'index.html'),
+    ...Object.fromEntries(
+      pages.map((page) => [page, path.join(__dirname, 'pages', page)]),
+    ),
+  },
+};
```

Comment:
This isn't needed. You can setup something simpler since we just want to build 1 page. In react-science we needed to build multiple pages.